packages/casting/src/follower-cosmjs.js

```diff
@@ -210,71 +201,64 @@ export const makeCosmjsFollower = (
       arrayEqual(result.subarray(0, dataPrefixBytes.length), dataPrefixBytes),
       X`${result} doesn't start with data prefix ${dataPrefixBytes}`,
     );
-    return result.slice(dataPrefixBytes.length);
-  };
-
-  /**
-   * @param {number} [height]
-   * @returns {Promise<Uint8Array>}
-   */
-  const getProvenDataAtHeight = async height => {
-    return retryGetDataAndStripPrefix(
-      async (endpoint, storeName, storeSubkey) => {
-        const queryClient = await provideQueryClient(endpoint);
-        return E(queryClient).queryVerified(storeName, storeSubkey, height);
-      },
-    );
+    // truncate 'value' of response
+    return { ...response, value: result.slice(dataPrefixBytes.length) };
   };
 
   /**
-   * @param {number} height
+   * @param {number} [height] desired height, or the latest height if not set
+   * @param {boolean} [prove=false]
+   * @returns {Promise<QueryAbciResponse>}
    */
-  const getUnprovenDataAtHeight = async height => {
+  const getAvailableData = async (height, prove = false) => {
     return retryGetDataAndStripPrefix(
       async (endpoint, storeName, storeSubkey) => {
         const client = await provideTendermintClient(endpoint);
         const response = await client.abciQuery({
           path: `store/${storeName}/key`,
           data: storeSubkey,
           height,
-          prove: false,
+          prove,
```

Comment:
I don’t think `abciQuery` checks the proofs, just includes the data necessary to verify the proofs in the response. I suspect we still need Stargate client to do the math on the proofs to verify them. But, this touches on stuff @michaelfig is far more knowledgeable about. https://github.com/cosmos/cosmjs/blob/74346b2ce6/packages/tendermint-rpc/src/tendermint34/requests.ts#L72-L78

Reply:
Ah, indeed. https://github.com/cosmos/cosmjs/blob/74346b2ce6/packages/stargate/src/queryclient/queryclient.ts#L508

I'll start over the refactoring without removing this.